cmd/lotus-shed/sectors.go

```diff
@@ -563,3 +571,220 @@ func rleToPng(rleBytes []byte) ([]byte, error) {
+
+		// Setup remote sector store
+		sminfo, err := lcli.GetAPIInfo(cctx, repo.StorageMiner)
+		if err != nil {
+			return xerrors.Errorf("could not get api info: %w", err)
+		}
+
+		localStore, err := paths.NewLocal(ctx, &emptyLocalStorage{}, api, []string{})
+		if err != nil {
+			return err
+		}
+
+		remote := paths.NewRemote(localStore, api, sminfo.AuthHeader(), 10,
+			&paths.DefaultPartialFileHandler{})
+
+		readStarter, err := remote.Reader(ctx, sref, abi.PaddedPieceSize(offset), abi.PaddedPieceSize(length))
+		if err != nil {
+			return xerrors.Errorf("getting reader: %w", err)
+		}
+
+		rd, err := readStarter(0)
+		if err != nil {
+			return xerrors.Errorf("starting reader: %w", err)
+		}
+
+		upr, err := fr32.NewUnpadReaderBuf(rd, abi.PaddedPieceSize(length), make([]byte, 1<<20))
+		if err != nil {
+			return xerrors.Errorf("creating unpadded reader: %w", err)
+		}
+
+		l := int64(abi.PaddedPieceSize(length).Unpadded())
+
+		bar := pb.New64(l)
+		br := bar.NewProxyReader(upr)
+		bar.ShowTimeLeft = true
+		bar.ShowPercent = true
+		bar.ShowSpeed = true
+		bar.Units = pb.U_BYTES
+		bar.Output = os.Stderr
+		bar.Start()
+
+		_, err = io.CopyN(os.Stdout, br, l)
+		if err != nil {
+			return xerrors.Errorf("reading data: %w", err)
+		}
+
+		return nil
+	},
+}
+
+var sectorDeleteCmd = &cli.Command{
+	Name:  "delete",
+	Usage: "delete a sector file from sector storage",
+	Flags: []cli.Flag{
+		&cli.BoolFlag{
+			Name: "really-do-it",
+		},
+	},
+	ArgsUsage: "[sector num] [file type]",
+	Action: func(cctx *cli.Context) error {
+		if cctx.NArg() != 2 {
+			return xerrors.Errorf("must pass sectornum/filetype")
+		}
+
+		sectorNum, err := strconv.ParseUint(cctx.Args().Get(0), 10, 64)
+		if err != nil {
+			return xerrors.Errorf("parsing sector number: %w", err)
+		}
+
+		var ft storiface.SectorFileType
+		switch cctx.Args().Get(1) {
+		case "cache":
+			ft = storiface.FTCache
+		case "sealed":
+			ft = storiface.FTSealed
+		case "unsealed":
+			ft = storiface.FTUnsealed
+		case "update-cache":
+			ft = storiface.FTUpdateCache
+		case "update":
+			ft = storiface.FTUpdate
+		default:
+			return xerrors.Errorf("invalid file type")
+		}
+
+		ctx := lcli.ReqContext(cctx)
+		api, closer, err := lcli.GetStorageMinerAPI(cctx)
+		if err != nil {
+			return err
+		}
+		defer closer()
+
+		maddr, err := api.ActorAddress(ctx)
+		if err != nil {
+			return xerrors.Errorf("getting miner actor address: %w", err)
+		}
+
+		mid, err := address.IDFromAddress(maddr)
+		if err != nil {
+			return xerrors.Errorf("getting miner id: %w", err)
+		}
+
+		sid := abi.SectorID{
+			Miner:  abi.ActorID(mid),
+			Number: abi.SectorNumber(sectorNum),
+		}
+
+		// get remote store
+		sminfo, err := lcli.GetAPIInfo(cctx, repo.StorageMiner)
+		if err != nil {
+			return xerrors.Errorf("could not get api info: %w", err)
+		}
+
+		localStore, err := paths.NewLocal(ctx, &emptyLocalStorage{}, api, []string{})
+		if err != nil {
+			return err
+		}
+
+		if !cctx.Bool("really-do-it") {
```

Comment:
We have this on most irreversible commands. It's there so that it's possible to run most of the command checks without performing the irreversible part. It also signals to the users that the command can perform dangerous actions which can cause problems.